### DISCUSSION TITLE
Can't push image to GPR from Actions

### DISCUSSION THREAD

**Discussion by augustocdias** [community/community#26232] [Actions]
<p>I am trying to push a docker image to GPR but it is failing </p>
<p>I’m using this to login (took from the dockerpush template)</p>
<pre><code>name: build-base-images
on:
  push:
    paths:
      - Dockerfile.base
      - Dockerfile.vips
      - Makefile
      - .github/workflows/base_image.yml

jobs:
  build_base_docker_image:
    runs-on: ubuntu-18.04
    steps:
      - uses: actions/checkout@v2
      - name: Docker login
        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login docker.pkg.github.com -u ${{ github.actor }} --password-stdin
      - name: Build and Publish to GPR
        run: make build-base-image
</code></pre>
<p>It is failing to bush the built image with this error:</p>
<pre><code>Successfully tagged docker.pkg.github.com/olxgroup-oss/dali/base-dali:c58f694
docker tag "docker.pkg.github.com/olxgroup-oss/dali/base-rust-image:c58f694" "docker.pkg.github.com/olxgroup-oss/dali/base-rust-image:latest"
docker push "docker.pkg.github.com/olxgroup-oss/dali/base-rust-image:c58f694"
The push refers to repository [docker.pkg.github.com/olxgroup-oss/dali/base-rust-image]
denied: Resource not accessible by integration
</code></pre>
<p>Do I really have to create a personal token for it?</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/augustocdias">@augustocdias</a> ,</p>
<p>According to official <a href="https://help.github.com/en/packages/using-github-packages-with-your-projects-ecosystem/configuring-docker-for-use-with-github-packages#authenticating-with-a-personal-access-token" rel="nofollow noopener">doc</a>, github token is permission enough to publish docker images to GPR.</p>
<p>I forked your repo, github token works fine on my side, it’s successful to publish docker image to GPR. Please check my repo <a href="https://github.com/weide-zhou/dali/blob/460ef98cb19360b8fce2e71324515648bcff8d5b/.github/workflows/base_image.yml#L18" rel="nofollow noopener">here</a>.</p>
<p>It’s recommended to have a double check with Github token, thanks.</p>

**Comment by augustocdias** [comment]
<p>I had the same exact code and if you take a look on the history from the actions in my repo you’ll see it is failing. Could it be because my repo is in a org which I’m not the owner and yours is in your personal account? In the repo I have <code>Admin</code> role, in the org I’m a <code>Member</code></p>

**Comment by ethomson** [comment, marked as answer]
<p>Hi <a class="mention" href="/augustocdias">@augustocdias</a> - two questions:</p>
<ol>
<li>
<p>Can you try your publish again?  There was a transient problem with uploading packages using the included GitHub token last week.  Your problem may be resolved.</p>
</li>
<li>
<p>If not, are you publishing the package within the same repository as the workflow that you’re running?  (Your logs mention that the image is tagged olxgroup-oss/dali - is this the repository that’s running the workflow?  If not, you will need to use a personal access token, since the GitHub token is scoped to a single repository.</p>
</li>
</ol>

**Comment by augustocdias** [comment]
<p>Thank you. </p>
<p>It worked indeed. </p>
